### PR TITLE
feat: preview config

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Read from admin.preview configuration and validate its contents.

The configuration format will need to be documented in our official docs.

### CODE REVIEW

**Comment by Marc-Roig** [comment]
@remidej let me know what you think on the overall structure! I am going to be creating unit tests & updating the example projects

File: packages/core/content-manager/server/src/preview/services/__tests__/preview-config.test.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  return {
+    enabled,
+    config: {
+      handler,
+    },
+  };
+};
+
+describe('Preview Config', () => {
+  test('Is not enabled by default', async () => {
```
Comment (remidej) [inline packages/core/content-manager/server/src/preview/services/__tests__/preview-config.test.ts:13]: it doesn't look like async is necessary in these

File: packages/core/content-manager/server/src/preview/services/preview-config.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      if (!this.isEnabled()) {
+        return;
+      }
+
+      const handler = this.getPreviewHandler();
+
+      // Handler must be a function
+      if (typeof handler !== 'function') {
+        throw new errors.ValidationError(
+          'Static Preview configuration is invalid. Handler must be a function'
```
Comment (remidej) [inline packages/core/content-manager/server/src/preview/services/preview-config.ts:45]: I would just say preview because it soon will no longer be static

**Review by remidej — COMMENTED** [review]
looks good!

**Review by remidej — APPROVED** [review]

**Pull request merged**